templates/default/machine/init.d.erb

```diff
@@ -18,18 +18,20 @@ fi
 INSTALL_DIR="<%= @install_dir %>"
 RETVAL=0
 PROG="$INSTALL_DIR/run.sh"
+pid_file="<%= @pid_file %>"
 
 start() {
     echo -n $"Starting AppDynamics machine agent:"
-    "$PROG"
+    daemon --pidfile=${pid_file} ${PROG}
```

Comment:
start_daemon is not available on CentOS where I'm testing, so I used daemon.
I've done a 'kitchen test machine-agent-centos' and it passes for both 6.7 and 7.2 I don't have a RHEL available, so I can't speak to that.